### ISSUE TITLE
ModelSEED SBML model can't be read due to duplicated IDs

### ISSUE THREAD

**Issue by dengxiao01** [opencobra/cobrapy#1235]
#### Problem description

![image](https://user-images.githubusercontent.com/103829632/174247098-e506c98a-d370-499f-af2f-2b5bda9b07e9.png)
![image](https://user-images.githubusercontent.com/103829632/174247253-f5abfb59-56e7-4f8a-a669-00bcbfb35dcf.png)

#### Code Sample

Create a [minimal, complete, verifiable example](https://stackoverflow.com/help/mcve).

```python
```

```
```

### Context

<details>

```
```

</details>

**Comment by matthiaskoenig** [comment]
What is the result of the SBML validation? I am pretty sure your SBML is invalid and you should fix the issues in your model file.

**Comment by dengxiao01** [comment]
> What is the result of the SBML validation? I am pretty sure your SBML is invalid and you should fix the issues in your model file.

SBML validation of web site, http://sbml.org/validator, is invalid, 404, is there where you can verify？

**Comment by matthiaskoenig** [comment]
Sorry, link changed with the site migration.
https://sbml.org/validator_servlet/
Best Matthias

**Comment by matthiaskoenig** [comment]
Fixed in https://github.com/opencobra/cobrapy/pull/1239

**Comment by Midnighter** [comment]
Any chance that you can add a redirect on the proxy that runs sbml.org?

**Comment by dengxiao01** [comment]
> Sorry, link changed with the site migration. https://sbml.org/validator_servlet/ Best Matthias
thanks,this is the result of the SBML validation:
This document is not valid SBML!
Error[ Line 1181](https://sbml.org/validator_servlet/section-print-result.jsp#line-1181) Column 54: (SBML Validation Rule #10301) The value of the 'id' field on every instance of the following type of object in a model must be unique: <model>, <functionDefinition>, <compartmentType>, <compartment>, <speciesType>, <species>, <reaction>, <speciesReference>, <modifierSpeciesReference>, <event>, and model-wide <parameter>s. Note that <unitDefinition> and parameters defined inside a reaction are treated separately. Reference: L2V1 Section 3.5 The <species> id 'M_cpd02701_b' conflicts with the previously defined <species> id 'M_cpd02701_b' at line 674.
<species id="M_cpd02701_b" name="S_Adenosyl_4_methylthio_2_oxobutanoate_c0_b" compartment="c0" charge="0" boundaryCondition="true"/>

![image](https://user-images.githubusercontent.com/103829632/174700597-d850e71d-a137-45f1-a73e-49a14573263a.png)

**Comment by dengxiao01** [comment]
> Any chance that you can add a redirect on the proxy that runs sbml.org?

Thank you. How do I do that? I'm not adding weighted direction

**Comment by Midnighter** [comment]
> > Any chance that you can add a redirect on the proxy that runs sbml.org?
> 
> Thank you. How do I do that? I'm not adding weighted direction

Sorry, that was a comment for @matthiaskoenig

**Comment by cdiener** [comment]
Here is a MRE:

1. Go to https://modelseed.org/genomes/Plants and download the A. Thaliana (2nd in the list) SBML model.
2. Read the model with `cobra.io.read_sbml_model`

This will result in:

```
[...]
ValueError: id 'cpd02701_b' at index 1154 is non-unique. Is it present twice?

The above exception was the direct cause of the following exception:

CobraSBMLError                            Traceback (most recent call last)
Input In [8], in <cell line: 1>()
----> 1 mod = read_sbml_model("Athaliana-TAIR10.sbml")

File ~/code/cobrapy/src/cobra/io/sbml.py:472, in read_sbml_model(filename, number, f_replace, **kwargs)
    462 except Exception as original_error:
    463     cobra_error = CobraSBMLError(
    464         "Something went wrong reading the SBML model. Most likely the SBML"
    465         " model is not valid. Please check that your model is valid using "
   (...)
    470         "at https://github.com/opencobra/cobrapy/issues ."
    471     )
--> 472     raise cobra_error from original_error

CobraSBMLError: Something went wrong reading the SBML model. Most likely the SBML model is not valid. Please check that your model is valid using the `cobra.io.sbml.validate_sbml_model` function or via the online validator at http://sbml.org/validator .
	`(model, errors) = validate_sbml_model(filename)`
If the model is valid and cannot be read please open an issue at https://github.com/opencobra/cobrapy/issues .
```

Which points towards the same issue as the SBML validator. The issue is that ModelSEED provided invalid SBML models (duplicated IDs). @Midnighter Do you know where we should report this issue and what ModelSEED uses to generate the SBML model?

**Comment by matthiaskoenig** [comment]
@Midnighter I just opened the issue for the redirect. This should be fixed soon. See https://github.com/sbmlteam/sbml-org-website/issues/39

**Comment by dengxiao01** [comment]
I tried to read it without error using libsbml：
reader = libsbml.SBMLReader()
document = reader.readSBML(sbml_test_file)
print(document.getNumErrors())
print(document.printErrors())
but the ModelSEED provided invalid SBML models (duplicated IDs),How does cobrapy solve this problem？

**Comment by matthiaskoenig** [comment]
Reading the document with libsbml does not perform validation. You have to run the validation before you will see the actual errors. LibSBML will provide exactly the same errors (i.e. you model is invalid because you have duplicate species in it), because cobrapy and the sbml.org validator are using libsbml.
In summary, your model is invalid and you have to fix the model. This is not related to cobrapy but to ModelSEED creating/distributing invalid/broken models. You have to fix your duplicate species. This can be complicated, because both will most likely appear in reactions and you have to resolve which of the species belongs actual to which reaction.
In summary: this is not an issue of cobrapy, but of your model.

**Comment by cdiener** [comment]
Closing in favor of the attached ModelSEED issue where that issue originates.
